testsuite/MDAnalysisTests/utils/test_units.py

```diff
@@ -83,37 +81,51 @@ def check_physical_constant(name, reference):
 class TestConversion(object):
     @staticmethod
     def _assert_almost_equal_convert(value, u1, u2, ref):
-        assert_almost_equal(units.convert(value, u1, u2), ref,
+        val = units.convert(value, u1, u2)
+        assert_almost_equal(val, ref,
                             err_msg="Conversion {0} --> {1} failed".format(u1, u2))
 
-    # generate individual test cases using nose's test generator mechanism
-    def test_length(self):
-        nm = 12.34567
-        A = nm * 10.
-        yield self._assert_almost_equal_convert, nm, 'nm', 'A', A
-        yield self._assert_almost_equal_convert, A, 'Angstrom', 'nm', nm
-
-    def test_time(self):
-        yield self._assert_almost_equal_convert, 1, 'ps', 'AKMA', 20.45482949774598
-        yield self._assert_almost_equal_convert, 1, 'AKMA', 'ps', 0.04888821
-
-    def test_energy(self):
-        yield self._assert_almost_equal_convert, 1, 'kcal/mol', 'kJ/mol', 4.184
-        yield self._assert_almost_equal_convert, 1, 'kcal/mol', 'eV', 0.0433641
-
-    def test_force(self):
-        yield self._assert_almost_equal_convert, 1, 'kJ/(mol*A)', 'J/m', 1.66053892103219e-11
-        yield self._assert_almost_equal_convert, 2.5, 'kJ/(mol*nm)', 'kJ/(mol*A)', 0.25
-        yield self._assert_almost_equal_convert, 1, 'kcal/(mol*Angstrom)', 'kJ/(mol*Angstrom)', 4.184
-
-    @staticmethod
-    def test_unit_unknown():
-        nm = 12.34567
-        assert_raises(ValueError, units.convert, nm, 'Stone', 'nm')
-        assert_raises(ValueError, units.convert, nm, 'nm', 'Stone')
-    
-    @staticmethod
-    def test_unit_unconvertable():
+    nm = 12.34567
```

Comment:
This is an example of using `parametrize` not `yield`, generally the idiom transforms like this:
```python
def test_this():
    for a, b in ((1, 2), (3, 4)):
        yield self._check, a, b

@pytest.mark.parametrize('a,b', ((1, 2), (3, 4)))
def test_this(a, b):
    self._check(a, b)
```

So it's mostly just that the `for` loop moves to before the test function rather than inside